test/Label.test.js

```diff
@@ -418,6 +435,686 @@ describe('Network Label', function() {
+        nodes: new vis.DataSet(dataNodes),
+        edges: []
+    };
+  
+    var options = {
+      nodes: {
+        font: {
+          multi: true
+        }
+      },
+      groups: {
+        group1: {
+          font: { color: 'red' },
+        },
+        group2: {
+          font: { color: 'white' },
+        },
+      },
+    };
+  
+    var network = new vis.Network(container, data, options);
+
+
+    assert.equal(modBold(0).color, '#343434');  // Default value
+    assert.equal(modBold(1).color, '#343434');  // Default value
+    assert.equal(modBold(2).color, 'red');      // Group value overrides default
+    assert.equal(modBold(3).color, 'green');    // Local value overrides default
+    assert.equal(modBold(4).color, 'green');    // Local value overrides group
+
+
+    //
+    // Change some node values dynamically
+    //
+    data.nodes.update([
+      {id: 1, group: 'group2'},
+      {id: 4, font: { bold: { color: 'orange'}}},
+    ]);
+
+    assert.equal(modBold(0).color, '#343434');  // unchanged
+    assert.equal(modBold(1).color, 'white');    // new group value
+    assert.equal(modBold(3).color, 'green');    // unchanged
+    assert.equal(modBold(4).color, 'orange');   // new local value
+
+    //
+    // Change group options dynamically
+    //
+    network.setOptions({
+      groups: {
+        group1: {
+          font: { color: 'brown' },
+        },
+      },
+    });
+
+
+    assert.equal(modBold(0).color, '#343434');  // unchanged
+    assert.equal(modBold(1).color, 'white');    // Unchanged
+    assert.equal(modBold(2).color, 'brown');    // New group values
+    assert.equal(modBold(3).color, 'green');    // unchanged
+    assert.equal(modBold(4).color, 'orange');   // unchanged
+
+
+    network.setOptions({
+      nodes: {
+        font: {
+          multi: true,
+          bold: {
+            color: 'black'
+          }
+        }
+      },
+    });
+
+    assert.equal(modBold(0).color, 'black');    // nodes default
+    assert.equal(modBold(1).color, 'black');    // more specific bold value overrides group value
+    assert.equal(modBold(2).color, 'black');    // idem
+    assert.equal(modBold(3).color, 'green');    // unchanged
+    assert.equal(modBold(4).color, 'orange');   // unchanged
+
+
+    network.setOptions({
+      groups: {
+        group1: {
+          font: { bold: {color: 'brown'} },
+        },
+      },
+    });
+
+    assert.equal(modBold(0).color, 'black');    // nodes default
+    assert.equal(modBold(1).color, 'black');    // more specific bold value overrides group value
+    assert.equal(modBold(2).color, 'brown');    // bold group value overrides bold node value
+    assert.equal(modBold(3).color, 'green');    // unchanged
+    assert.equal(modBold(4).color, 'orange');   // unchanged
+
+
+    //
+    // Same initialization as previous with a color set for the default node font
+    //
+    data.nodes = new vis.DataSet(dataNodes);  // Need to reset nodes, changed in previous
```

Comment:
This isn't really a reset is it?

It's a new dataset, but the since dataNodes is a mutable array of mutable objects, and not a function who's return value is a new array of objects, the nodes have not been reset.

Reply:
Yes, but `dataNodes` is not changed anywhere, so the same original values are being entered in the `DataSet` instance, and `DataSet` copies values internally.

Crap, let me confirm that.